test_util/aws.py

```diff
@@ -127,11 +127,36 @@ def wait_for_stack_deletion(self, wait_before_poll_min=3):
         self.wait_for_status_change('DELETE_IN_PROGRESS', 'DELETE_COMPLETE', wait_before_poll_min)
 
     def get_parameter(self, param):
+        """Returns param if in stack parameters, else returns None
+        """
         for p in self.stack.parameters:
             if p['ParameterKey'] == param:
                 return p['ParameterValue']
         return None
 
+    def get_resource_by_type(self, resource_type):
+        """For a full list of AWS resource types, see:
+        http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html  # noqa
+        """
+        return [r for r in self.stack.resource_summaries.all()
+                if r.resource_type == resource_type]
+
+    @retry_boto_rate_limits
+    def get_auto_scaling_instances(self, logical_id):
+        """ Get instances in ASG with logical_id. If logical_id is None, all ASGs will be used
+        Will return instance objects as describd here:
+        http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#instance
+        """
+        if logical_id is None:
+            asg_list = [asg.physical_resource_id for asg in
```

Comment:
See below. This was motivated from the confusing naming of the VPC ASG and then it just allowed me to have a slightly cleaner looking implementation in the AdvancedCf, but I am fine with changing to have more hard failures.